webhook-service/lib/request_validator.go

```diff
@@ -0,0 +1,44 @@
+package lib
+
+import (
+	"fmt"
+	"strings"
+)
+
+type requestValidator struct {
+	denyListProvider DenyListProvider
+	ipResolver       IPResolver
+}
+
+type RequestValidator interface {
+	Validate(request Request) error
+}
+
+func NewRequestValidator(denyListProvider DenyListProvider, ipResolver IPResolver) RequestValidator {
+	validator := requestValidator{
+		denyListProvider: denyListProvider,
+		ipResolver:       ipResolver,
+	}
+	return validator
+}
+
+func (c requestValidator) Validate(request Request) error {
+	if request.URL == "" {
+		return fmt.Errorf("invalid curl URL: '%s'", request.URL)
```

Comment:
since `request.URL` will always be just empty:
```suggestion
		return fmt.Errorf("curl command contains empty URL")
```